google/cloud/sql/connector/instance_connection_manager.py

```diff
@@ -287,6 +327,25 @@ def _auth_init(self) -> None:
 
         self._credentials = credentials
 
+    async def seconds_until_refresh(self, expiration: datetime.datetime) -> int:
```

Comment:
Does this need to be async? I don't see any awaits happening inside it.

Reply:
Originally we were awaiting the metadata in the async function to calculate the expiration, but then that was moved so we calculate the seconds_until_refresh when creating the metadata. 
Now we're awaiting the metatata in this function again so it needs to be async.